kong/plugins/bot-detection/handler.lua

```diff
@@ -31,30 +125,24 @@ function BotDetectionHandler:access(conf)
     return responses.send_HTTP_BAD_REQUEST(err)
   end
 
-  if user_agent then
-    user_agent = strip(user_agent)
+  if not user_agent then
+    return
+  end
 
-    if conf.whitelist then
-      for _, rule in ipairs(conf.whitelist) do
-        if re_match(user_agent, rule) then
-          return
-        end
-      end
-    end
+  local match
+  local api_id = ngx.ctx.api.id
+  if not ua_cache[api_id][user_agent] then
+    match = examine_agent(user_agent, conf)
+    ua_cache[api_id][user_agent] = match
 
-    if conf.blacklist then
-      for _, rule in ipairs(conf.blacklist) do
-        if re_match(user_agent, rule) then
-          return responses.send_HTTP_FORBIDDEN()
-        end
-      end
-    end
-  
-    for _, rule in ipairs(rules.bots) do
-      if re_match(user_agent, rule) then
-        return responses.send_HTTP_FORBIDDEN()
-      end
-    end
+  else
```

Comment:
(pedantic style: as per our code style, we are missing a blank line above this token)